src/components/CheckboxFieldItem/CheckboxFieldItem.tsx

```diff
@@ -79,8 +85,8 @@ export const CheckboxFieldItem = ({
   /**
    * Check for previous props/states
    */
-  function usePrevious(checkboxChecked: any) {
-    const ref = useRef();
+  function usePrevious(checkboxChecked: boolean) {
+    const ref = useRef() as MutableRefObject<boolean>;
```

Comment:
```suggestion
    const ref = useRef<boolean>();
```

Where possible, it's better to use a type param, to prevent mistakes/shenanigans like

```ts
const ref = useRef('hi') as MutableRefObject<boolean>; // TS is fine with this

const ref = useRef<boolean>('hi'); // ERROR! TS saves us from ourselves, once again
```

Reply:
ah TIL, thanks. I was following the [guidelines BM added](https://github.com/chanzuckerberg/edu-design-system/blob/next/docs/CODE_GUIDELINES.md#useref) 😅  so will update those as well

Reply:
Actually, technically it should be

```ts
const ref = useRef<boolean | undefiend>();
```

since we're using an initial value of `undefined`.

OR, use `false` as the initial value

```suggestion
    const ref = useRef(false);
```

and then there's no need to even specify the type (since it will be inferred)/